### PR TITLE
Add AllNoFailMethod

### PR DESCRIPTION
**Pull request by johnnorton**
AllNoFail (except will fail if everything fails :) )  Not sure I love the method name....let me know

### CODE REVIEW

**Comment by johnnorton** [comment]
I had to add some JavaDoc to get the build to run...

**Comment by tylerhjones** [comment]
Try not to format the license blocks, makes it look like the wording has changed.

File: src/main/java/com/jive/foss/pnky/Pnky.java [hunk 1]
```diff
@@ -1048,11 +1065,117 @@ public void onFailure(@Nonnull final Throwable t)
...
+      promise.alwaysAccept(new ThrowingBiConsumer<V, Throwable>()
+      {
+        @Override
+        public void accept(final V result, final Throwable error) throws Throwable
+        {
+          if (null != result)
+          {
+            actualResultsCount.incrementAndGet();
+          }
+          else if (null != error)
```
Comment (bpedman) [inline src/main/java/com/jive/foss/pnky/Pnky.java]: this if/else block should really just be:

```
if (null != error)
{
  actualErrorCount.incrementAndGet();
}
else
{
  actualResultsCount.incrementAndGet();
}
```

A promise may be successfully completed with a null value (and is actually the case quite a bit in code I have written).

File: src/main/java/com/jive/foss/pnky/PnkyPromise.java [hunk 2]
```diff
@@ -46,7 +42,7 @@
  *     recover the result.</li>
  *     <li>Handle a transformed result that returns the result of another asynchronous action.</li>
  *   </ol>
- * </p>
```
Comment (bpedman) [inline src/main/java/com/jive/foss/pnky/PnkyPromise.java]: I don't believe this should be deleted

File: src/main/java/com/jive/foss/pnky/PnkyPromise.java [hunk 3]
```diff
@@ -17,18 +17,14 @@
...
 import java.util.concurrent.Executor;
 
-import com.google.common.util.concurrent.Futures;
-import com.google.common.util.concurrent.ListenableFuture;
-import com.jive.foss.commons.function.ThrowingBiConsumer;
-import com.jive.foss.commons.function.ThrowingBiFunction;
-import com.jive.foss.commons.function.ThrowingConsumer;
-import com.jive.foss.commons.function.ThrowingFunction;
-import com.jive.foss.commons.function.ThrowingRunnable;
-
```
Comment (bpedman) [inline src/main/java/com/jive/foss/pnky/PnkyPromise.java]: Undo the change to the imports, the style is to not use * imports except for static imports

File: src/main/java/com/jive/foss/pnky/Pnky.java [hunk 4]
```diff
@@ -17,34 +17,26 @@
...
+import lombok.NonNull;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
```
Comment (bpedman) [inline src/main/java/com/jive/foss/pnky/Pnky.java]: Undo the change to the imports, the style is to not use * imports except for static imports

File: src/main/java/com/jive/foss/pnky/FullResult.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+ * Created by jonorton on 6/6/16.
+ *
+ * Result class used to review results from {@link Pnky#full(Iterable)} method.
+ */
+@AllArgsConstructor
+@Getter
+public class FullResult<V>
+{
+  private final List<V> results;
+  private final Throwable throwable;
```
Comment (bpedman) [inline src/main/java/com/jive/foss/pnky/FullResult.java]: This should be a `List<Throwable>` which I can use to map the exception to the original promise I submitted.

File: src/main/java/com/jive/foss/pnky/Pnky.java [hunk 6]
```diff
@@ -1048,11 +1065,117 @@ public void onFailure(@Nonnull final Throwable t)
...
+
+            if (actualResultsCount.get() == 0)
+            {
+              // Nothing worked, everything failed! we need to complete exceptionally
+              pnky.reject(combinedException);
+            }
+            else
+            {
+              // There is a combination of results and maybe some errors
+              pnky.resolve(new FullResult<V>(Arrays.asList(results), combinedException));
```
Comment (bpedman) [inline src/main/java/com/jive/foss/pnky/Pnky.java]: See https://github.com/jive/pnky-promise/pull/1/files#r67464667 the combined exception here will be replaced with just `Arrays.asList(errors)`

File: src/main/java/com/jive/foss/pnky/Pnky.java [hunk 7]
```diff
@@ -1048,11 +1065,117 @@ public void onFailure(@Nonnull final Throwable t)
...
+      final Iterable<? extends PnkyPromise<? extends V>> promises)
+  {
+    final Pnky<FullResult<V>> pnky = Pnky.create();
+
+    final int numberOfPromises = Iterables.size(promises);
+
+    // Special case, no promises to wait for
+    if (numberOfPromises == 0)
+    {
+      return Pnky.immediatelyComplete(new FullResult<V>(Collections.<V> emptyList(), null));
```
Comment (bpedman) [inline src/main/java/com/jive/foss/pnky/Pnky.java]: replace null with `Collections.emptyList()` as well after changing the field

**Comment by bpedman** [comment]
I would also like to see a simple test case added for this
